site/en/blog/tether-elements-to-each-other-with-css-anchor-positioning/index.md

```diff
@@ -0,0 +1,831 @@
+
+const anchor = document.querySelector('.anchor')
+const tooltip = document.querySelector('.tooltip')
+
+const updatePosition = () => {  
+  computePosition(anchor, tooltip, {
+    placement: 'top',
+    middleware: [offset(10), flip()]
+  })
+    .then(({x, y}) => {
+      Object.assign(tooltip.style, {
+        left: `${x}px`,
+        top: `${y}px`
+      })
+  })
+};
+
+const clean = autoUpdate(anchor, tooltip, updatePosition);
+```
+
+Try re-positioning the anchor in this demo that uses that code.
+
+{% Codepen {
+    user: 'web-dot-dev',
+    id: 'wvEaOrP',
+    height: 450,
+    tab: 'result'
+  }
+%}
+
+The "tooltip" might not behave how you expect. It reacts to going outside of the viewport on the y-axis but not the x-axis. Dig into the [documentation](https://floating-ui.com/docs/flip), and you’ll likely find a solution that works for you.
+
+But, finding a package that works for your project can take a lot of time. It's extra decisions and can be frustrating if it doesn't quite work how you want.
+
+## Using anchor positioning
+Enter the CSS anchor positioning API. The idea is to keep your styles in your CSS and reduce the number of decisions you need to make. You're hoping to achieve the same result, but the goal is to make the developer experience better.
+
+- No JavaScript required.
+- Let the browser work out the best position from your guidance.
+- No more third party dependencies
+- No wrapper elements.
+- Works with elements that are in the top layer.
+
+Let's recreate and tackle the problem we were trying to solve above. But, instead, use the analogy of a boat with an anchor. These represent the anchored element and anchor. The water represents the containing block.
+
+First, you need to choose how to define the anchor. You can do this within your CSS by setting the `anchor-name` property on the anchor element. It accepts a [dashed-ident](https://drafts.csswg.org/css-values-4/#typedef-dashed-ident) value.
+
+``` css
+.anchor {
+  anchor-name: --my-anchor;
+}
+```
+
+Alternatively, you will be able to define an anchor in your HTML with the `anchor` attribute. The attribute value is the ID of the anchor element. This creates an implicit anchor.
+
+``` html
+<a id="my-anchor" class="anchor"></a>
+<div anchor="my-anchor" class="boat">I’m a boat!</div>
+```
+
+Once you've defined an anchor, you can use the `anchor` function. The `anchor` function takes 3 arguments:
+
+**Anchor element:** The `anchor-name` of the anchor to use—or, you can omit the value to use an `implicit` anchor. It can be defined via the HTML relationship, or with an `anchor-default` property with an `anchor-name` value.
+**Anchor side:** A keyword of the position you want to use. This could be `top`, `right`, `bottom`, `left`, `center`, etc. Or, you can pass a percentage. For example, 50% would be equal to `center`.
+**Fallback:** This is an optional fallback value that accepts a length or percentage.
```

Comment:
```suggestion
- **Anchor element:** The `anchor-name` of the anchor to use—or, you can omit the value to use an `implicit` anchor. It can be defined via the HTML relationship, or with an `anchor-default` property with an `anchor-name` value.
- **Anchor side:** A keyword of the position you want to use. This could be `top`, `right`, `bottom`, `left`, `center`, etc. Or, you can pass a percentage. For example, 50% would be equal to `center`.
- **Fallback:** This is an optional fallback value that accepts a length or percentage.
```